tools/jdk/local_java_repository.bzl

```diff
@@ -34,10 +46,39 @@ def _local_java_repository_impl(repository_ctx):
     )
 
     extension = ".exe" if repository_ctx.os.name.lower().find("windows") != -1 else ""
-    if java_home_path.get_child("bin").get_child("java" + extension).exists:
+    java_bin = java_home_path.get_child("bin").get_child("java" + extension)
+    if java_bin.exists:
+        version = repository_ctx.attr.version if repository_ctx.attr.version != "" else _detect_java_version(repository_ctx, java_bin)
+
         repository_ctx.file(
             "BUILD.bazel",
-            repository_ctx.read(repository_ctx.path(repository_ctx.attr._build_file)),
+            repository_ctx.read(repository_ctx.path(repository_ctx.attr._build_file)) +
+            """
+config_setting(
+    name = "name_setting",
+    values = {{"java_runtime_version": "{local_jdk}"}},
+    visibility = ["//visibility:private"],
+)
+config_setting(
+    name = "version_setting",
+    values = {{"java_runtime_version": "{version}"}},
+    visibility = ["//visibility:private"],
+)
+alias(
```

Comment:
We already have skylib in Bazel core, use [`selects.with_or`](https://github.com/bazelbuild/bazel-skylib/blob/master/docs/selects_doc.md#selectswith_or) instead of rolling your own alias?

Reply:
`local_java_repository` is also used in `jdk.WORKSPACE` suffix. I would like to avoid importing skylib there.